Sources/Compiler/IR/Module.swift

```diff
@@ -39,6 +39,24 @@ public struct Module {
   /// The module's name.
   public var name: String { program.ast[decl].name }
 
+  /// Accesses the function at given index.
+  public subscript(function index: Functions.Index) -> Function {
+    _read   { yield  functions[index] }
+    _modify { yield &functions[index] }
+  }
+
+  /// Accesses the basic block with the given identity.
+  public subscript(block id: Block.ID) -> Block {
+    _read   { yield  functions[id.function].blocks[id.address] }
+    _modify { yield &functions[id.function].blocks[id.address] }
+  }
```

Comment:
```suggestion
  /// Accesses the given block.
  public subscript(_ b: Block.ID) -> Block {
    _read   { yield  functions[b.function].blocks[b.address] }
    _modify { yield &functions[b.function].blocks[b.address] }
  }
```

Totally instinct question, and I'll accept a one-word answer: do we really want all of these subscripts to be public?

Reply:
yes